In [71]:
# Importing Libs
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import pylab as pl
%matplotlib inline

In [120]:
# Importing Data
df = pd.read_csv("C:\\Users\\Yasser Arafat\\Desktop\\FuelConsumptionCo2.csv") 
cdf = df[['ENGINESIZE','CYLINDERS','FUELCONSUMPTION_CITY','FUELCONSUMPTION_HWY','FUELCONSUMPTION_COMB','CO2EMISSIONS']]
# cdf.head(9)

In [109]:
msk = np.random.rand(len(df)) < 0.8
train = cdf[msk]
test = cdf[~msk]

In [119]:
from sklearn import linear_model
regr = linear_model.LinearRegression()
x = np.asanyarray(train[['ENGINESIZE','CYLINDERS','FUELCONSUMPTION_HWY','FUELCONSUMPTION_CITY']])
y = np.asanyarray(train[['CO2EMISSIONS']])
regr.fit (x, y)
# The coefficients
print ('Coefficients: ', regr.coef_[0])
print("Intercept : ",regr.intercept_)

Coefficients:  [9.99183346 7.28894822 3.01955001 6.54685754]
Intercept :  [64.61700957]


In [125]:
y_results= regr.predict(test[['ENGINESIZE','CYLINDERS','FUELCONSUMPTION_HWY','FUELCONSUMPTION_CITY']])
x = np.asanyarray(test[['ENGINESIZE','CYLINDERS','FUELCONSUMPTION_HWY','FUELCONSUMPTION_CITY']])
y = np.asanyarray(test[['CO2EMISSIONS']])
print("Residual sum of squares: %.2f" % np.mean((y_results - y) ** 2))
# Explained variance score: 1 is perfect prediction
print('Variance score: %.2f' % regr.score(x, y))

Residual sum of squares: 528.67
Variance score: 0.88


In [128]:
#Comparing Predited Values with Original Data
for i in range(int(len(y_results)*0.2)):
    print("{} compared with {} ".format(round(y_results[i][0],0),y[i][0]))

256.0 compared with 255 
325.0 compared with 354 
367.0 compared with 359 
248.0 compared with 251 
288.0 compared with 288 
218.0 compared with 242 
229.0 compared with 246 
313.0 compared with 336 
205.0 compared with 209 
216.0 compared with 237 
380.0 compared with 380 
202.0 compared with 200 
180.0 compared with 181 
199.0 compared with 193 
205.0 compared with 209 
205.0 compared with 209 
241.0 compared with 237 
218.0 compared with 213 
290.0 compared with 281 
246.0 compared with 246 
295.0 compared with 292 
220.0 compared with 191 
259.0 compared with 278 
199.0 compared with 212 
214.0 compared with 218 
210.0 compared with 221 
217.0 compared with 235 
216.0 compared with 232 
254.0 compared with 248 
332.0 compared with 340 
273.0 compared with 294 
259.0 compared with 262 
313.0 compared with 292 
353.0 compared with 382 
186.0 compared with 192 
185.0 compared with 177 
339.0 compared with 380 
388.0 compared with 344 
340.0 compared with 380 
373.0 compared with 435 


# Thus the Accuracy of our prediction model is 88%, and this 16% more accurate than our Simple Linear Regression of the same Data Set.